processor/tailsamplingprocessor/processor.go

```diff
@@ -296,26 +296,75 @@ func (tsp *tailSamplingSpanProcessor) ConsumeTraces(ctx context.Context, td pdat
 	return nil
 }
 
-func (tsp *tailSamplingSpanProcessor) groupSpansByTraceKey(resourceSpans pdata.ResourceSpans) map[pdata.TraceID][]*pdata.Span {
-	idToSpans := make(map[pdata.TraceID][]*pdata.Span)
-	ilss := resourceSpans.InstrumentationLibrarySpans()
+func (tsp *tailSamplingSpanProcessor) groupResourceSpansByTraceID(rsSource pdata.ResourceSpans) map[pdata.TraceID]*pdata.ResourceSpans {
+	// here, we need to do two things: find out how many traces are within the resource spans
+	// so that we return a map based on the trace ID
+	// and then we iterate again to create one resource span per trace ID
+	idToResourceSpans := make(map[pdata.TraceID]*pdata.ResourceSpans)
+
+	ilss := rsSource.InstrumentationLibrarySpans()
 	for j := 0; j < ilss.Len(); j++ {
-		spans := ilss.At(j).Spans()
+		// first, keep track of the trace IDs we have in the batch
+		ilsSource := ilss.At(j)
+		spans := ilsSource.Spans()
 		spansLen := spans.Len()
 		for k := 0; k < spansLen; k++ {
 			span := spans.At(k)
-			key := span.TraceID()
-			idToSpans[key] = append(idToSpans[key], &span)
+			rs := pdata.NewResourceSpans()
+			idToResourceSpans[span.TraceID()] = &rs
 		}
 	}
-	return idToSpans
+
+	// now, group the spans on resource spans based on the original ones,
+	// split by trace ID
+	for traceID := range idToResourceSpans {
+		rs := idToResourceSpans[traceID]
+		rsSource.Resource().MoveTo(rs.Resource())
```

Comment:
Shouldn't this be `CopyTo`? Otherwise only the first resource spans will be filled

Reply:
I see the mismatch: the problem is that I'm considering only one ResourceSpan per trace ID, which is clearly wrong. I need to go back to the drawing board.

Reply:
This actually begs the question: how to group the resource spans? Or should we group them at all? I think the grouping of it was discussed as part of another issue ("compacting" was the term used there, I believe)

cc @pkositsyn, @pmm-sumo

Reply:
I suggest adding something like `Equal` method for `Resource` and `InstrumentationLibrary`. As for me, this is the most accurate and safe approach (for example if new fields are added to `Resource`, then it will be easy to notice that equal func should be updated). But anyway it's possible to compare `Resouce` manually via `Range` of `AttributesMap`, so equality function can also be added right here

Reply:
That's actually a good idea. What do you tink, @bogdandrutu, @tigrannajaryan and @codeboten?

Reply:
I think `Equal` would be nice to have. For grouping under the common `Resource`/`InstrumentationLibrary` either `groupbyattrsprocessor` could be used separately (in "compaction" mode) or maybe it could be done altogether in this code to reduce the number of calls and simplify configuration